pkg/controller/operators/olm/operator.go

```diff
@@ -1685,7 +1685,11 @@ func (a *Operator) updateInstallStatus(csv *v1alpha1.ClusterServiceVersion, inst
 	}
 
 	if strategyErr != nil {
-		csv.SetPhaseWithEventIfChanged(requeuePhase, requeueConditionReason, fmt.Sprintf("installing: %s", strategyErr), now, a.recorder)
+		if install.ReasonForError(strategyErr) == install.StrategyErrDeploymentUpdated {
+			csv.SetPhaseWithEventIfChanged(v1alpha1.CSVPhaseInstallReady, requeueConditionReason, fmt.Sprintf("installing: %s", strategyErr), now, a.recorder)
+		} else {
+			csv.SetPhaseWithEventIfChanged(requeuePhase, requeueConditionReason, fmt.Sprintf("installing: %s", strategyErr), now, a.recorder)
```

Comment:
Is the reason `InstallWaiting` when this branch is hit, and is that the most appropriate reason in this case?

Reply:
This is somewhat weird given the function signature - you would expect the requeue phase to be used.

Reply:
Discussed offline - this whole state management should probably be refactored to make this behavior clearer, but doing so here would make it difficult to backport.